### DISCUSSION TITLE
A great library - Language support future. - Filament

### DISCUSSION THREAD

**Discussion by atul161** [donmccurdy/glTF-Transform#763] [General]
Hi @donmccurdy ,

First of all thankyou for creating this library! This library is very critical for us
and you guys have done tremendous job. Lots of Appreciation.

Currently i am using filament in our android code base and i don't found any open source project that perform
so much operation on gltf as your library do. Just wanted to know your approach. 

Is there any future support for kotlin so mobile devs can easily able to use.

Even i can use this library in our android project also using js interface(Android ie Android js). 
but i want to get stream of new glb so that i can manage caching and storage.

Waiting for your view! Thanks!

**Comment by donmccurdy** [comment]
Hello @atul161, and thank you!

My intention is that glTF Transform should work in any JavaScript runtime (web, node.js, deno, bun...). The first two are well-supported, but for deno and bun I don't really have the infrastructure and resources to test right now. And unfortunately, I don't know anything at all about running a JavaScript runtime in an Android project. :)

I don't have any plans of porting this library to another language like Kotlin, but for anyone interested in something like that, I'd refer you to my recent comments here:

- https://github.com/donmccurdy/glTF-Transform/discussions/714

You may also want to have a look at [cgltf](https://github.com/jkuhlmann/cgltf), if you haven't already. It's used internally in Filament, and presumably can be used on Android. It doesn't provide higher-level operations like glTF Transform's [/functions](https://gltf-transform.donmccurdy.com/functions.html), but it does provide a useful API for reading/writing glTF data.
